OMCompiler/Compiler/BackEnd/BackendDAEOptimize.mo

```diff
@@ -2420,6 +2420,11 @@ algorithm
     // true case left with condition<>false
     case (_,_,_,_,_,_,_)
       equation
+        // ticket #13927
+        // temporarily disable this simplification as it is not correct.
+        // TODO fix the simplification
+        false = true;
```

Comment:
that code line really raises my eyebrows 👀 ^^ 
Is that some Metamodelica construct? Because on an innocent glance this looks like it should be a compilation error (or part of a code obfuscation contest) 😝

Reply:
An equality that's false causes the code to fail, which works a bit like `throw` in e.g. C++. In this case it just jumps to the next case in the `matchcontinue` statement. Just calling `fail()` would perhaps be a bit more idiomatic, or just commenting out the whole case.

Reply:
True, my apologies, I wanted a quick fix for the issue and didn't try too hard to make it pretty. Anyways this PR needs a little more work so I will fix this as well.